kedro/framework/cli/catalog.py

```diff
@@ -179,6 +179,18 @@ def create_catalog(metadata: ProjectMetadata, pipeline_name, env):
         click.echo("All datasets are already configured.")
 
 
+@catalog.command("factory list")
+@env_option
+@click.pass_obj
+def list_patterns(metadata: ProjectMetadata, env):
+    "Show all factory patterns in the catalog, ranked by priority by which they are matched."
+    session = _create_session(metadata.package_name, env=env)
+    context = session.load_context()
+
+    catalog_factories = context.catalog._dataset_patterns
+    click.echo(yaml.dump(list(catalog_factories.keys())))
```

Comment:
```suggestion
    dataset_patterns = context.catalog._dataset_patterns
    click.echo(yaml.dump(list(dataset_patterns.keys())))
```
minor nitpick since we are using `_dataset_patterns` might be more fitting to call this `dataset_patterns`.

Reply:
I think there will still be some inconsistency, with the change we will still be dumping "dataset patterns" under the name catalog factories 🤔. How about `catalog_factory_patterns`? A tad verbose but more readable/understandable.

Reply:
Added clarification to the docstring, and some comments on consistency to the PR description. Let me know if you think the variables should still be updated!

Reply:
Happy to leave as is.